client/server/pages/index.js

```diff
@@ -52,20 +52,24 @@ const debug = debugFactory( 'calypso:pages' );
 
 const calypsoEnv = config( 'env_id' );
 
+let branchName;
 function getCurrentBranchName() {
-	try {
-		return execSync( 'git rev-parse --abbrev-ref HEAD' ).toString().replace( /\s/gm, '' );
-	} catch ( err ) {
-		return undefined;
+	if ( ! branchName ) {
+		try {
+			branchName = execSync( 'git rev-parse --abbrev-ref HEAD' ).toString().replace( /\s/gm, '' );
```

Comment:
This is definitely a separate PR material. Nice optimization btw 👍

Reply:
Sure thing 👍

Reply:
https://github.com/Automattic/wp-calypso/pull/66517